File: examples/test-chat.js
#!/usr/bin/env node

/**
 * Venice.ai API - Chat Completion Test
 * 
 * This script tests the basic chat completion endpoint.
 * 
 * Usage:
 *   export VENICE_API_KEY=your-api-key-here
 *   node test-chat.js
 */

const VENICE_API_KEY = process.env.VENICE_API_KEY;

if (!VENICE_API_KEY) {
  console.error('❌ Error: VENICE_API_KEY environment variable is not set');
  console.error('');
  console.error('Please set your API key:');
  console.error('  export VENICE_API_KEY=your-api-key-here');
  console.error('');
  console.error('Get your API key from: https://venice.ai/dashboard');
  process.exit(1);
}

async function testChatCompletion() {
  console.log('🚀 Testing Venice.ai Chat Completion API...\n');

  try {
    const response = await fetch('https://api.venice.ai/api/v1/chat/completions', {
      method: 'POST',
      headers: {
        'Authorization': `Bearer ${VENICE_API_KEY}`,
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        model: 'venice-uncensored',
        messages: [
          {
            role: 'user',
            content: 'Hello! Say "Venice API is working!" if you can hear me.'
          }
        ],
        temperature: 0.7,
        max_completion_tokens: 100
      })
    });

    if (!response.ok) {
      const error = await response.json();
      console.error('❌ API Error:', error.error?.message || JSON.stringify(error));
      process.exit(1);
    }

    const data = await response.json();
    
    console.log('✅ Success! Response received:\n');
    console.log('Model:', data.model);
    console.log('Content:', data.choices[0].message.content);
    console.log('\nUsage:');
    console.log('  Prompt tokens:', data.usage.prompt_tokens);
    console.log('  Completion tokens:', data.usage.completion_tokens);
    console.log('  Total tokens:', data.usage.total_tokens);
    console.log('\n🎉 Venice.ai API is working correctly!');

  } catch (error) {
    console.error('❌ Error:', error.message);
    process.exit(1);
  }
}

testChatCompletion();

File: examples/test-chat.py
#!/usr/bin/env python3

"""
Venice.ai API - Chat Completion Test

This script tests the basic chat completion endpoint.

Usage:
    export VENICE_API_KEY=your-api-key-here
    python3 test-chat.py
"""

import os
import sys
import requests
import json

VENICE_API_KEY = os.getenv('VENICE_API_KEY')

if not VENICE_API_KEY:
    print('❌ Error: VENICE_API_KEY environment variable is not set', file=sys.stderr)
    print('', file=sys.stderr)
    print('Please set your API key:', file=sys.stderr)
    print('  export VENICE_API_KEY=your-api-key-here', file=sys.stderr)
    print('', file=sys.stderr)
    print('Get your API key from: https://venice.ai/dashboard', file=sys.stderr)
    sys.exit(1)

def test_chat_completion():
    print('🚀 Testing Venice.ai Chat Completion API...\n')

    try:
        response = requests.post(
            'https://api.venice.ai/api/v1/chat/completions',
            headers={
                'Authorization': f'Bearer {VENICE_API_KEY}',
                'Content-Type': 'application/json',
            },
            json={
                'model': 'venice-uncensored',
                'messages': [
                    {
                        'role': 'user',
                        'content': 'Hello! Say "Venice API is working!" if you can hear me.'
                    }
                ],
                'temperature': 0.7,
                'max_completion_tokens': 100
            },
            timeout=30
        )

        if not response.ok:
            try:
                error = response.json()
                error_field = error.get("error", {})
                if isinstance(error_field, dict):
                    error_msg = error_field.get("message", json.dumps(error))
                else:
                    error_msg = str(error_field)
            except Exception:
                error_msg = f"HTTP {response.status_code}: {response.text}"
            print(f'❌ API Error: {error_msg}', file=sys.stderr)
            sys.exit(1)

        data = response.json()
        
        print('✅ Success! Response received:\n')
        print(f'Model: {data["model"]}')
        print(f'Content: {data["choices"][0]["message"]["content"]}')
        print('\nUsage:')
        print(f'  Prompt tokens: {data["usage"]["prompt_tokens"]}')
        print(f'  Completion tokens: {data["usage"]["completion_tokens"]}')
        print(f'  Total tokens: {data["usage"]["total_tokens"]}')
        print('\n🎉 Venice.ai API is working correctly!')

    except requests.exceptions.RequestException as error:
        print(f'❌ Error: {error}', file=sys.stderr)
        sys.exit(1)
    except Exception as error:
        print(f'❌ Error: {error}', file=sys.stderr)
        sys.exit(1)

if __name__ == '__main__':
    test_chat_completion()
